mirdata/guitarset.py

```diff
@@ -0,0 +1,391 @@
+        )
+        self.audio_hex_path = os.path.join(
+            self._data_home, self._track_paths['audio_hex'][0]
+        )
+        self.audio_mic_path = os.path.join(
+            self._data_home, self._track_paths['audio_mic'][0]
+        )
+        self.audio_mix_path = os.path.join(
+            self._data_home, self._track_paths['audio_mix'][0]
+        )
+        self.jams_path = os.path.join(self._data_home, self._track_paths['jams'][0])
+
+        title_list = track_id.split('_')  # [PID, S-T-K, mode, rec_mode]
+        style, tempo, _ = title_list[1].split('-')  # [style, tempo, key]
+        self.player_id = title_list[0]
+        self.mode = title_list[2]
+        self.tempo = float(tempo)
+        self.style = _STYLE_DICT[style[:-1]]
+
+    # this lets users run `print(Track)` and get actual information
+    def __repr__(self):
+        repr_string = "GuitarSet Track(track_id={}, jams_path={})"
+        return repr_string.format(self.track_id, self.jams_path)
+
+    # `annotation` will behave like an attribute, but it will only be loaded
+    # and saved when someone accesses it. Useful when loading slightly
+    # bigger files or for bigger datasets. By default, we make any time
+    # series data loaded from a file a cached property
+    @utils.cached_property
+    def beats(self):
+        return _load_beats(self.jams_path)
+
+    @utils.cached_property
+    def leadsheet_chords(self):
+        if self.mode == 'solo':
+            logging.info(
+                'Chord annotations for solo excerpts are the same with the comp excerpt.'
+            )
+        return _load_chords(self.jams_path, leadsheet_version=True)
+
+    @utils.cached_property
+    def infered_chords(self):
+        if self.mode == 'solo':
+            logging.info(
+                'Chord annotations for solo excerpts are the same with the comp excerpt.'
+            )
+        return _load_chords(self.jams_path, leadsheet_version=False)
+
+    @utils.cached_property
+    def key_mode(self):
+        return _load_key_mode(self.jams_path)
+
+    @utils.cached_property
+    def pitch_contours(self):
+        contours = []
+        # iterate over 6 strings
+        for i in range(6):
+            contours.append(_load_pitch_contour(self.jams_path, i))
+        return contours
+
+    @utils.cached_property
+    def notes(self):
+        notes = []
+        # iterate over 6 strings
+        for i in range(6):
+            notes.append(_load_note_ann(self.jams_path, i))
+        return notes
+
+    # `audio` will behave like an attribute, but it will only be loaded
+    # when someone accesses it and it won't be stored. By default, we make
+    # any memory heavy information (like audio) properties
+    @property
+    def audio(self):
+        """Load the default audio version (mic) of the GuitarSet Track.
+        """
+        return self.load_audio()
+
+    def load_audio(self, version='mic'):
```

Comment:
For consistency with the rest of the library, can you remove this function, and create specific `Track` properties for each audio type (`audio_mix`, `audio_hex`, etc)? The functionality you have in each branch of `load_audio` can be called directly, e.g. 

```python
    @property
    def audio_mix(self):
        """Load the mix audio version of the GuitarSet Track.
        """
        return librosa.load(self.audio_mix_path, sr=None)
```

Reply:
OK, thanks! Was wondering what's the best way to deal with multiple audio files without breaking the quasi-API